topk-js/src/collection.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{
+  document::{DocumentWrapper, Value},
+  query::Query,
+};
+use napi::bindgen_prelude::*;
+use napi_derive::napi;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+#[napi]
+pub struct CollectionClient {
+  collection: String,
+  client: Arc<topk_rs::Client>,
+}
+
+#[napi]
+impl CollectionClient {
+  pub fn new(client: Arc<topk_rs::Client>, collection: String) -> Self {
+    Self { client, collection }
+  }
+
+  #[napi]
+  pub async fn query(&self, query: Query, lsn: Option<u32>) -> Result<Vec<HashMap<String, Value>>> {
+    let docs = self
+      .client
+      .collection(&self.collection)
+      .query(query.into(), lsn.map(|l| l as u64), None)
+      .await
+      .map_err(|e| napi::Error::new(napi::Status::GenericFailure, e.to_string()))?;
+
+    Ok(
+      docs
+        .into_iter()
+        .map(|d| DocumentWrapper::from(d).into())
+        .collect(),
+    )
+  }
+
+  #[napi]
+  pub async fn upsert(&self, docs: Vec<HashMap<String, Value>>) -> Result<i32> {
```

Comment:
nit: we return `lsn` as `i32` here and as `i64` on the method below

Reply:
Updated it to `i64`, however, `u64` is not supported by NAPI by default. Check out [numbers](https://napi.rs/docs/concepts/values#numbers) section.
`u64` is supported via their [`BigInt`](https://napi.rs/docs/concepts/values#bigint) value which is a custom type in Typescript and imo feels a bit off since javascript user would just expect a `number` type.

Reply:
yeah, this sucks... it makes me think that the client actually doesn't care about the value, it only uses `lsn` as a "snapshot of database". we could encode the strings using https://docs.rs/bs58/latest/bs58/. its base-58 strings that look nicely and random. wdyt? that way we'd return a string here, same in python, same in rust

Reply:
no strong opinions about this, string makes my life easier so if it makes sense all across the board then why not